include/crocoddyl/core/state-base.hxx

```diff
@@ -50,6 +50,8 @@ template <typename Scalar>
 std::vector<typename MathBaseTpl<Scalar>::MatrixXs> StateAbstractTpl<Scalar>::Jdiff_Js(
     const Eigen::Ref<const VectorXs>& x0, const Eigen::Ref<const VectorXs>& x1, Jcomponent firstsecond) {
   MatrixXs Jfirst(ndx_, ndx_), Jsecond(ndx_, ndx_);
+  Jfirst.setZero();
```

Comment:
We do not need `setZero()`, could you remove these lines?.
Note that the matrix are always constructed with zeros.

In Crocoddyl we don't initialize them with `nan` as it could happen in Pinocchio.
And all data classes have been designed with this in mind.

Reply:
Jfirst and Jsecond are input from the user. If we want the python bindings to work, these have to be set to zero. They are not set to zero by default.

Reply:
`Jfirst` and `Jsecond` aren't in/out arguments in this function (`Jdiff_Js`). They are constructed inside and you can see in line 52.

It is an alternative function to `Jdiff`, which it is the one that you referred to.

Reply:
From the Eigen Documentation at https://eigen.tuxfamily.org/dox/group__TutorialMatrixClass.html:

**Constructors**
---------------------------------------

Constructors taking sizes are also available. For matrices, the number of rows is always passed first. For vectors, just pass the vector size. They allocate the array of coefficients with the given size, **but don't initialize the coefficients themselves:**
```
MatrixXf a(10,15);
VectorXf b(30);
```
Here,

- a is a 10x15 dynamic-size matrix, with allocated but **currently uninitialized coefficients**.
- b is a dynamic-size vector of size 30, with allocated but **currently uninitialized coefficients**.
-----------------------------------------------------------------

Also, the unittests sometimes fail because of this :)

Reply:
I'll resolve this.